src/logic/wallets/store/middlewares/providerWatcher.ts

```diff
@@ -16,50 +19,60 @@ export const loadLastUsedProvider = async (): Promise<string | undefined> => {
   return lastUsedProvider
 }
 
-let watcherInterval
-const providerWatcherMware = (store) => (next) => async (action) => {
-  const handledAction = next(action)
+type ProviderWatcherAction = {
+  type: string
+  payload: ProviderState
+}
 
-  if (watchedActions.includes(action.type)) {
-    switch (action.type) {
-      case ADD_PROVIDER: {
-        const currentProviderProps = action.payload.toJS()
+let watcherInterval: NodeJS.Timer
+const providerWatcherMware =
+  (store: typeof reduxStore) =>
```

Comment:
Here's an example of the type you removed being used again. Maybe it should instead be `ReturnType<typeof reduxStore>`?

Reply:
It's your own commit that I pulled from main.

Reply:
Ok, I will add these comments to the Redux store typing issue then.